modules/mono/editor/bindings_generator.cpp

```diff
@@ -1945,6 +1969,12 @@ Error BindingsGenerator::_generate_cs_method(const BindingsGenerator::TypeInterf
 	const TypeInterface *return_type = _get_type_or_null(p_imethod.return_type);
 	ERR_FAIL_NULL_V(return_type, ERR_BUG); // Return type not found
 
+	if (return_type->is_singleton) {
+		StringName instance_type_name = return_type->name + CS_SINGLETON_INSTANCE_SUFFIX;
+		return_type = &obj_types.find(instance_type_name)->value;
+		ERR_FAIL_NULL_V(return_type, ERR_BUG); // Return type not found
+	}
```

Comment:
This block exists 5 times so it should probably be a dedicated method.

Reply:
Honestly, I'm not sure it's worth it, it would only save 3 lines. Do you think it makes sense to add it to `_get_type_or_null` instead? Since it seems this block is always done after calling that one if the resolved type has `is_singleton` set to true. Not sure if there's any case where we don't want to get the singleton type despite `is_singleton` being true which is what made me hesitant to do this in the first place.